usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -343,6 +343,29 @@ def construct_es_response_for_prime_awards(self, response) -> dict:
             if row.get("Awarding Agency"):
                 code = row.pop("agency_code")
                 row["awarding_agency_id"] = self.get_agency_database_id(code)
+            if row.get("COVID-19 Obligations"):
+                row["COVID-19 Obligations"] = sum(
+                    [
+                        x["obligation"]
+                        if (self.filters.get("def_codes") is not None and x["defc"] in self.filters["def_codes"])
+                        or self.filters.get("def_codes") is None
+                        else 0
+                        for x in row.get("COVID-19 Obligations")
+                    ]
+                )
+            if row.get("COVID-19 Outlays"):
+                row["COVID-19 Outlays"] = sum(
+                    [
+                        x["outlay"]
+                        if (self.filters.get("def_codes") is not None and x["defc"] in self.filters["def_codes"])
+                        or self.filters.get("def_codes") is None
+                        else 0
+                        for x in row.get("COVID-19 Outlays")
+                    ]
+                )
+            if row.get("def_codes"):
+                if self.filters.get("def_codes"):
+                    row["def_codes"] = filter(lambda x: x in self.filters.get("def_codes"), row["def_codes"])
```

Comment:
You might need to wrap the `filter()` in `list()` because testing locally it appears that DRF is representing a filter object as an empty list.

Reply:
Also, do we want to limit the DEFC returned to only those that are filtered on? Realized while working on the download counterpart that it wasn't mentioned in any AC and keep going back and forth on it. Originally was for limiting the DEFC shown to only those that are filtered to, but could come across as hiding information. My thinking is that Outlay and Obligation are different since on that page we don't list everything out, only a single number (unlike the list of DEFC next to it).

Reply:
Initially I was not going. to filter the DEFC array, but based on [this](https://data-transparency-bfs.slack.com/archives/C0122HUCD17/p1620831604086100) conversation with Beth, it seems that the PO wants the array to be filtered as well. I also am not seeing the bug you're talking about with the `def_codes` array, but I'll try and look into it some more.

Reply:
>Initially I was not going. to filter the DEFC array, but based on this conversation with Beth, it seems that the PO wants the array to be filtered as well

Gotcha. I will make the change to the download as well to reflect this change.